static_src/components/form/form_element.jsx

```diff
@@ -11,77 +12,62 @@ export default class FormElement extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      err: null,
+      err: null, // Forms should all be initialized without errors
       value: this.props.value || ''
     };
 
     if (!this.props.key) {
       this.state.id = nextId();
     }
 
-    this.componentWillMount = this.componentWillMount.bind(this);
-    this.componentWillUnmount = this.componentWillUnmount.bind(this);
     this.onChange = this.onChange.bind(this);
   }
 
-  componentWillMount() {
-    if (this.props.attachToForm) {
-      this.props.attachToForm(this);
-    }
-  }
-
-  componentDidMount() {
-    this.validate();
-  }
-
   componentWillReceiveProps(props) {
     // If our validator changed, we want to update the error state.
     // TODO we probably also want to notify others about the validation result,
     // but this is a poor place to do it.
-    const err = props.validator(this.state.value, props.label);
-    this.setState({ err });
-  }
-
-  componentWillUnmount() {
-    if (this.props.detatchFromForm) {
-      this.props.detatchFromForm(this);
+    if (this.props.validator !== props.validator) {
+      const err = props.validator(this.state.value, props.label);
+      this.setState({ err, value: props.value });
     }
   }
 
   onChange(e) {
-    this.setState({ value: e.target.value }, () => {
-      this.validate();
+    const value = e.target.value;
+    const err = this.props.validator(value, this.props.name);
+    this.setState({ err, value }, () => {
+      formActions.changeField(this.props.formGuid, this.props.name, value)
+        .then(() => {
```

Comment:
Ah because the `err` is here. So this code will set the current state of the form, so now it renders with the new value and an error. Then it will change the field so the form model in the store is updated. Then it will set an error or success of the field through the form actions?